### PR TITLE
WIP CMake - fixes #13

### LINKED ISSUE #13: Перевести проект на систему сборки cmake
Или любую кроссплатформенную систему сборки

### PR DESCRIPTION
**Pull request by IonAgorria**
This a work in progress changes to add CMake support to allow more portable/flexible build system and potentially support more platforms in future.

On Linux Boost dependency detection works, not sure about Windows.

Needs to be tested in Windows, as code doesn't compile fully yet on other platforms.

### CODE REVIEW

**Comment by caiiiycuk** [comment]
Nice work :+1:

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+# depends
+FIND_PACKAGE(Boost)
+
+# globals
+INCLUDE_DIRECTORIES(SYSTEM
+        ${Boost_INCLUDE_DIRS}
+        )
+
+INCLUDE(CheckCXXCompilerFlag)
+SET(CMAKE_CXX_FLAGS_DEBUG "-g -Wall -Wextra")
```
Comment (SSE4) [inline CMakeLists.txt:28]: for sure will not work under MSVC, move under if

Comment (IonAgorria) [reply CMakeLists.txt]: Done, Im not sure if MSVC will need any specific stuff but IF is in place if thats the case

**Comment by DileSoft** [comment]
Предлагаю перенаправить в ветку cmake.

**Pull request merged**